staging/src/k8s.io/apimachinery/pkg/util/runtime/runtime.go

```diff
@@ -230,21 +232,22 @@ type rudimentaryErrorBackoff struct {
 	minPeriod time.Duration // immutable
 	// TODO(lavalamp): use the clock for testability. Need to move that
 	// package for that to be accessible here.
-	lastErrorTimeLock sync.Mutex
-	lastErrorTime     time.Time
+	lastErrorTime atomic.Pointer[time.Time]
 }
 
 // OnError will block if it is called more often than the embedded period time.
 // This will prevent overly tight hot error loops.
 func (r *rudimentaryErrorBackoff) OnError() {
-	now := time.Now() // start the timer before acquiring the lock
-	r.lastErrorTimeLock.Lock()
-	d := now.Sub(r.lastErrorTime)
-	r.lastErrorTime = time.Now()
-	r.lastErrorTimeLock.Unlock()
-
-	// Do not sleep with the lock held because that causes all callers of HandleError to block.
-	// We only want the current goroutine to block.
+	now := time.Now() // start the timer before atomic operations
+	var t time.Time
+	if loaded := r.lastErrorTime.Load(); loaded != nil {
+		t = *loaded
+	} else {
+		panic("rudimentaryErrorBackoff.lastErrorTime was not initialized")
+	}
```

Comment:
not sure I see the need to add a panic case here... if we ever encounter this, we can just leave `t` as zero-value, `d` will be large, we won't end up sleeping, and storing `r.lastErrorTime.Store(&now)` will make the next invocation work fine

Reply:
Good call. I've dropped the else and force pushed.